zotero/src/data_structure/shared_fields.rs

```diff
@@ -24,6 +23,32 @@ pub struct Link {
     pub r#type: String,
 }
 
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+pub struct Tag {
+    pub tag: String
+}
+
+pub trait Identifier {
+    fn key(&self) -> &String;
+}
+
+pub trait ItemCommon {
+    fn title(&self) -> &str;
+
+    fn tags(&self) -> &Vec<Tag>;
+
+    fn key(&self) -> &str;
+
+    fn has_tag(&self, tag: String) -> bool {
+        for t in self.tags() {
+            if tag == t.tag {
+                return true;
+            }
+        }
+        return false;
```

Comment:
Personally, I prefer a more functional style

```suggestion
      self.tags().iter().any(|t| t.tag == tag)
```